utils/spacewalk-utils.spec

```diff
@@ -20,90 +21,113 @@
+%else
+Requires:       perl(:MODULE_COMPAT_%(eval "`%{__perl} -V:version`"; echo $version))
+%endif
+# Required by debsolver.py
+Requires:       python3-PyYAML
+# Required by debsolver.py, cloneByDate.py
+Requires:       python3-uyuni-common-libs
+# Required by spacewalk-clone-by-date, spacewalk-sync-setup
+Requires:       salt
+# Required by cloneByDate.py, spacewalk-clone-by-date
+Requires:       spacewalk-backend
+# Required by cloneByDate.py
+Requires:       spacewalk-backend-sql
+# Required by spacewalk-final-archive
+Requires:       spacewalk-backend-tools >= 2.2.27
+# Required by cloneByDate.py, depsolver.py,spacewalk-clone-by-date
 Requires(pre):  uyuni-base-common
+
+%description
+Utilities that may be run against a SUSE Manager server (supported) or an Uyuni server
+
+%package extras
+Summary:        Extra utilities that may run against a SUSE Manager/Uyuni server
+# Required by spacewalk-dump-schema, spacewalk-host-rename (provides /usr/bin/spacewalk-sql)
+Requires:       susemanager-schema
+# Required by spacewalk-hostname-rename, spacewalk-watch-channel-sync.sh
 Requires:       bash
+# Required by spacewalk-hostname-rename
 Requires:       cobbler
-Requires:       coreutils
-Requires:       /usr/bin/spacewalk-sql
+# Required by spacewalk-hostname-rename
 Requires:       iproute
-Requires:       net-tools
+# Required by spacewalk-hostname-rename
 Requires:       perl-Satcon
+# Required by spacewalk-dump-schema, spacewalk-hostname-rename
 %if 0%{?suse_version}
 Requires:       perl = %{perl_version}
 %else
 Requires:       perl(:MODULE_COMPAT_%(eval "`%{__perl} -V:version`"; echo $version))
 %endif
-Requires:       python3
-Requires:       python3-rpm
+# Required by taskotop
+Requires:       python3-curses
+# Required by sw-ldap-user-sync
+Requires:       python3-ldap
+# Required by sw-ldap-user-sync
+Requires:       python3-PyYAML
+# Required by migrate-system-profile
 Requires:       python3-rhnlib >= 2.5.20
+# Required by migrateSystemProfile.py, spacewalk-common-channels, systemSnapshot.py
+Requires:       python3-uyuni-common-libs
+# spacewalk-dump-schema, spacewalk-hostname-rename
 Requires:       rpm
-Requires:       salt
+# Required by spacewalk-hostname-rename
 Requires:       spacewalk-admin
+# Required by spacewalk-common-channels, spacewalk-manage-snapshots, systemSnapshot.py
 Requires:       spacewalk-backend
-Requires:       python3-uyuni-common-libs
+# Required by taskotop
+Requires:       spacewalk-backend-sql
+# Required by spacewalk-final-archive, spacewalk-watch-channel-sync.sh
 Requires:       spacewalk-backend-tools >= 2.2.27
+# Required by spacewalk-hostname-rename
 Requires:       spacewalk-certs-tools
+# Required by spacewalk-hostname-rename
 Requires:       spacewalk-config
+# Required by spacewalk-export
 Requires:       spacewalk-reports
+# Required by spacewalk-hostname-rename
 Requires:       spacewalk-setup
+# As spacewalk-utils owns {python3_sitelib}/utils
+Requires:       spacewalk-utils
```

Comment:
If you require the main package, you should not add requires for things which are already in the main package.

Reply:
Done.